docs/services/postgresql/split-partitioned-database.md

```diff
@@ -0,0 +1,222 @@
+# Splitting a shard in postgresql
+
+## Assumptions
+
+`pg1` is currently the only database containing sharded data.
+Half of the data should be moved to a new `pg2` server
+
+Current database configuration:
+
+```python
+PARTITION_DATABASE_CONFIG = {
+    'shards': {
+        'p1': [0, 3],
+    },
+    'groups': {
+        'main': ['default'],
+        'proxy': ['proxy'],
+        'form_processing': ['p1'],
+    },
+}
+
+DATABASES = {
+    'proxy': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'commcarehq_proxy',
+        'USER': 'commcarehq',
+        'PASSWORD': 'commcarehq',
+        'HOST': 'pg1',
+        'PORT': '5432',
+        'TEST': {
+            'SERIALIZE': False,
+        },
+    },
+    'p1': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'commcarehq_p1',
+        'USER': 'commcarehq',
+        'PASSWORD': 'commcarehq',
+        'HOST': 'pg1',
+        'PORT': '5432',
+        'TEST': {
+            'SERIALIZE': False,
+        },
+    },
+}
+```
+
+## Process Overview
+
+1. Setup *pg2* node as a standby node of *pg1*
+2. Promote *pg2* to a master node
+3. Update the configuration so that requests for database alias *p2* go to *pg2* instead
+of *pg1*.
+4. Delete invalid data
+
+## Process detail
+
+### 1. Setup *pg2* as a standby node of *pg1*
+This step does not require downtime and can be done at any stage prior to the
+downtime.
+
+```bash
+commcare-cloud <env> ansible-playbook setup_pg_standby.yml -e standby=pg2 -e hot_standby_master=pg1 -e replication_slot=[replication slot name]
+```
+
+### 2. Stop all DB requests
+This will bring the CommCare HQ site down.
+
+**Stop all CommCare processes**
+```bash
+commcare-cloud <env> downtime start
+```
+
+**Stop pgbouncer**
+To be completely certain that no data will be updating during the move you can also
+prevent connections from pgbouncer:
+
+```bash
+pg1 $ psql -p 6432 -U someuser pgbouncer
+
+> PAUSE commcarehq_p1
+```
+
+### 3. Check document counts in the databases
+This step is useful for verifying the move at the end.
+```
+commcare-cloud <env> django-manage print_approximate_doc_distribution --csv
+```
+
+### 4. Update database name
+
+In pg2 postgres shell:
+
+```sql
+ALTER DATABASE commcarehq_p1 RENAME TO commcarehq_p2;
+```
```

Comment:
There's currently no way to manually specify a database name for a sharded database. It must be `commcarehq_django_alias`.

Reply:
sure but the name comes from the Django DB settings right? I mean I would still want to rename the DB.